seismostats/analysis/estimate_mc.py

```diff
@@ -342,82 +346,108 @@ def mc_by_bvalue_stability(
+        raise ValueError('Magnitudes are not binned correctly.')
+
+    if get_option("warnings") is True:
+        if not binning_test(mcs_test, delta_m, check_larger_binning=False):
+            warnings.warn("Mcs to test are not binned correctly. "
+                          "Test might fail because of this.")
+
+    b_values_test = []
     diff_bs = []
+    estimator = b_method()
+    value = False
+    best_mc = None
+    best_b_value = None
+
     for ii, mc in enumerate(mcs_test):
-        # TODO: here, one should be able to choose the method
-        b, std = estimate_b(
-            sample[sample >= mc - delta_m / 2], mc, delta_m,
-            return_std=True)
-        if len(sample[sample >= mc - delta_m / 2]) < 30:
-            warnings.warn(
-                "Number of events above tested Mc is less than 30. "
-                "This might affect the stability test."
-            )
-        bs.append(b)
+
+        if verbose:
+            print(f'\nTesting mc: {mc}.')
+
+        sample_selected = magnitudes[magnitudes >= mc - delta_m / 2]
+
+        if len(sample_selected) < 30:
+            warnings.warn("Number of events above tested Mc is less than 30. "
+                          "This might affect the stability test.")
+
+        estimator.calculate(sample_selected, mc, delta_m, **kwargs)
+        b = estimator.b_value
+        std = estimator.std
+        b_values_test.append(b)
 
         mc_plus = np.arange(mc, mc + stability_range, delta_m)
-        mc_plus = mc_plus[mc_plus <= np.max(sample)]
+        mc_plus = mc_plus[mc_plus <= np.max(magnitudes)]
         b_ex = []
         for mc_p in mc_plus:
-            # TODO: here, one should be able to choose the method
-            b_p = estimate_b(sample[sample >= mc_p - delta_m / 2],
-                             mc_p, delta_m)
-            b_ex.append(b_p)
+            sample_selected = magnitudes[magnitudes >= mc_p - delta_m / 2]
+            estimator.calculate(sample_selected, mc_p, delta_m, **kwargs)
+            b_ex.append(estimator.b_value)
         b_avg = np.sum(b_ex) / steps
         diff_b = np.abs(b_avg - b) / std
         diff_bs.append(diff_b)
+
+        if verbose:
+            print(f'..diff_b: {diff_b}')
+
         if diff_b <= 1:
             value = True
             if diff_b == min(diff_bs):
                 best_mc = mc
-                best_b = bs[-1]
+                best_b_value = b_values_test[-1]
             if stop_when_passed:
                 mcs_test = mcs_test[:ii + 1]
                 break
 
-    if value:
-        return bin_to_precision(best_mc, delta_m), best_b, mcs_test, bs, diff_bs
-    else:
-        raise ValueError("No Mc passes the stability test.")
+    if value and verbose:
+        print(f'\n\nBest mc to pass the test: {best_mc:.3f}',
+              f'\nwith a b-value of: {best_b_value:.3f}.')
+    elif verbose:
+        print("None of the mcs passed the stability test.")
+
```

Comment:
```suggestion
    elif not value:
        raise ValueError("None of the mcs passed the stability test.")
```
The "no mcs passes the test" should be raised in any case and not only if verbose is on